modules/rank-eval/src/main/java/org/elasticsearch/index/rankeval/RankEvalSpec.java

```diff
@@ -135,10 +162,44 @@ public void setSpecifications(Collection<RatedRequest> specifications) {
         } , REQUESTS_FIELD);
     }
 
+    public static RankEvalSpec parse(XContentParser parser, RankEvalContext context, boolean templated) throws IOException {
+        RankEvalSpec spec = PARSER.parse(parser, context);
+
+        logger.trace("request received: {}", spec.toString());
+        if (templated) {
+            String template = spec.getTemplate();
+            for (RatedRequest query_spec : spec.getSpecifications()) {
+                @SuppressWarnings({ "unchecked", "rawtypes" })
+                Map<String, String> params = (Map) query_spec.getParams();
+                Script script = new Script(template, ScriptService.ScriptType.INLINE, "mustache", params);
+                String resolvedRequest = 
```

Comment:
I think we should turn this code in a util method on ScriptService in core module. There are now several places where we have code similar to this. This would fix the code duplication.

Something like:

```java
public SearchSourceBuilder templateSearchRequest(Script script, QueryParseContext context) {
   ....
}
```